pdr-admin/src/app/protected-area/site/site-manage/site-edit-form/site-edit-form.component.ts

```diff
@@ -186,6 +187,12 @@ export class SiteEditFormComponent implements OnInit, OnDestroy {
 
   // empty function to allow use of shared template
   onDatePickerInteract() {
+    // TODO: Fix this bug
+    // For some reason, if we do detectChanages outisde of a setTimeout
+    // Month picker and year picker breaks
+    setTimeout(() => {
+      this.cdr.detectChanges();
```

Comment:
"this.cdr.detectChanges() only runs change detection for the current component (and descendants). If setFirstItemActive() causes changes elsewhere this is not covered. setTimeout() or zone.run(...) or ApplicationRef.tick() causes change detection to be run for the whole application and therefore every binding is covered, not only the current component."

![image](https://github.com/bcgov/parks-data-register/assets/10578223/8c02f9bf-3f1d-42c5-8df7-66c3585ed12e)

Reply:
![image](https://github.com/bcgov/parks-data-register/assets/71421099/8566a1fa-9026-4768-9592-948cf001f75b)

I tried the `ApplicationRef` methods and `setTimeout()` without `detectChanges` cuz I thought it felt cleaner but it didn't actually fix anything. Pretty sure when you and I ran into this problem with the datepicker for protected areas we ran into the same thing.

Its like Angular wants us to subscribe to `onChanges` and just run `detectChanges` every time something happens if we are running with lazy loading. He hmmmmmmmmmmmmmmmmmm